Hakawai/Core/HKWTextView.m

```diff
@@ -24,10 +24,7 @@
 @interface HKWTextView () <UITextViewDelegate, HKWAbstractionLayerDelegate, NSTextStorageDelegate>
 
 @property (nonatomic) NSMutableDictionary *simplePluginsDictionary;
-/*!
- String that saves the state of the text in the text view so that it can be accessed in the NSTextStorageDelegate, which will
- already have deleted the character by the time it's trying to process said deletion
- */
+
 @property (nonatomic, strong, readwrite) NSString *textStateBeforeDeletion;
```

Comment:
can we just synthesize instead of redeclaring?

Reply:
Synthesizing the readonly property would need us to access the property's ivar from methods outside init (in `padTextStorageForRangeInsertionAtLocation` and `textStorage:didProcessEditing`) while setting the property and this would be against linked-in guidelines(According to linked-in guidelines, ivars can be directly accessed when overriding getter/setter and in init and dealloc methods).

Having readwrite property in `HKWTextView.m` would be according to the guidelines and be accessible as read only outside the class.
Please let us know if it would still be better to synthesize the property.